src/haz3lcore/dynamics/Transition.re

```diff
@@ -394,16 +399,30 @@ module Transition = (EV: EV_MODE) => {
         }
       /* This case isn't currently used because deferrals are elaborated away */
       | DeferredAp(d3, d4s) =>
+        print_endline("DeferredAp in Ap");
+        print_endline("\t d2: " ++ DHExp.show(d2));
+        print_endline("\t d3: " ++ DHExp.show(d3));
+        print_endline("\t d4s: " ++ [%derive.show: list(DHExp.t)](d4s));
         let n_args =
           List.length(
-            List.map(
+            List.filter(
               fun
               | {term: Deferral(_), _} => true
               | _ => false: Exp.t => bool,
               d4s,
             ),
           );
-        let-unbox args = (Tuple(n_args), d2);
+        print_endline("\tn_args: " ++ string_of_int(n_args));
+        let-unbox args =
+          if (n_args == 1) {
+            (
+              Tuple(n_args),
+              Tuple([d2]) |> fresh // TODO Should we not be going to a tuple?
+            );
+          } else {
+            (Tuple(n_args), d2);
+          };
+        print_endline("\targs: " ++ [%derive.show: list(Exp.t)](args));
```

Comment:
I don't know if there's a better way to be handling this unboxing for the singleton case than wrapping it in a tuple.